src/root/components/draft/Checklist/index.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import InputLabel from '#components/draft/InputLabel';
+import InputError from '#components/draft/InputError';
+import List from '#components/draft/List';
+import Checkbox from '#components/draft/Checkbox';
+
+import styles from './styles.module.scss';
+
+type OptionKey = string | number;
+
+export interface Props<V, O, N> {
+  className?: string;
+  name: N;
+  options: O[];
+  onChange: (value: V[] | undefined, name: N) => void;
+  value: V[] | undefined;
+  label?: React.ReactNode;
+  hint?: React.ReactNode;
+  error?: string;
+  labelContainerClassName?: string;
+  hintContainerClassName?: string;
+  errorContainerClassName?: string;
+  checkboxListContainerClassName?: string;
+  keySelector: (option: O) => V;
+  labelSelector: (option: O) => React.ReactNode;
+  disabled?: boolean;
+  readOnly?: boolean;
+}
+
+function Checklist<
+  V extends OptionKey,
+  O extends object,
+  N extends string | number,
+>(props: Props<V, O, N>) {
```

Comment:
@frozenhelium I'll need a bit of help understanding some of this syntax

Reply:
Sure thing :)